### DISCUSSION TITLE
feature request: web: markdown editor: ADD upload image function with file selector (aside from drag&drop)

### DISCUSSION THREAD

**Discussion by allanlaal** [community/community#4040] [Discussions]
feature request: web: markdown editor: ADD upload image function with file selector (aside from drag&drop)

**Comment by allanlaal** [comment]
should insert image at cursor

**Reply by Chaostheorie** [reply]
Are you suggesting for every user to have a repository, e.g. 'images', or rather creating a gist per upload (file change)? Since this would only make a great difference for including images in issues (or discussions), since otherwise you can use relative paths to link repo files, such as images.

**Reply by laymonage** [reply]
@Chaostheorie no, the suggestion is to add an "upload image" button in the comment toolbar that lets you open the file selection dialog and upload images which will be inserted as markdown image links at the current cursor.

![image](https://user-images.githubusercontent.com/6379424/120009822-9b228680-c006-11eb-911f-2b59fe0fa52e.png)

Right now you can directly upload images to GitHub while writing comments by pasting an image or dragging and dropping the image to the comment box.

**Reply by Chaostheorie** [reply]
Ah, okay. I misread the request. Thank you for clarifying. This makes a lot more sense now.
